In [20]:
#t3
import math
data33 = [ 115, 125, 
102, 129, 121, 119, 120, 120, 
126, 120, 124, 118, 116, 132, 
114, 108, 127, 131, 130, 181]

odotusarvo = sum(data33)/len(data33)

z = (odotusarvo - 120)/(math.sqrt(60)/math.sqrt(len(data33)))
print(odotusarvo)
print(z)


123.9
2.2516660498395438


z0.975 < 2.2516660498395438 < z0.99

eli mu = 120 voidaan hyväksyä 0.005 merkittävyys tasolla.

In [21]:
#t4
data34 = [ 53.08, 57.07, 
54.10, 56.81, 58.85, 56.02, 59.08, 
55.78, 56.72, 54.92, 57.32, 59.00, 
54.91, 58.13, 60.69, 51.76, 52.31, 
60.50, 58.31, 58.70]

odotusarvo2 = sum(data34)/len(data34)

z2 = (odotusarvo2 - 55)/(2.3/math.sqrt(len(data34)))
print(odotusarvo2)
print(z)

56.702999999999996
2.2516660498395438


2.2516660498395438 > z0.975 hylätään

In [22]:
#t5
data35 = [ 37.5, 40.9, 43.3, 
37.2, 42.5, 42.0, 40.8, 40.9, 36.5, 
36.8, 39.7, 42.6, 45.9, 47.0, 33.8, 
40.8, 45.8, 36.5, 42.4, 46.4, 44.8, 
43.3, 53.7, 44.6, 45.6, 46.5, 36.8, 
47.8, 40.9, 48.6]

odotusarvo3 = sum(data35)/len(data35)

s = 0
for i in data35:
    s += (i - odotusarvo3)**2

s = s/(len(data35)-1)

t = (odotusarvo3 - 42.8)/(math.sqrt(s)/math.sqrt(len(data35)))

print(t)


-0.49731432337026826


jos merkitsevyystaso on 5% nolla hypoteesia ei hylätä sillä -t29,0.95 = - 1.699 < -0.49731432337026826